src/main/java/com/redhat/devtools/intellij/common/utils/ConfigWatcher.java

```diff
@@ -136,12 +128,13 @@ private void register(Path path, WatchService service, HighSensitivityRegistrar
                 service);
         }
 
-        private void watch(Consumer<Config> consumer, WatchService service) throws InterruptedException {
+        private void watch(Consumer<io.fabric8.kubernetes.client.Config> listener, WatchService service) throws InterruptedException {
             for (WatchKey key = service.take(); key != null; key = service.take()) {
                 key.pollEvents().forEach((event) -> {
                     Path changed = getAbsolutePath(directory, (Path) event.context());
                     if (isConfigPath(changed)) {
-                        consumer.accept(loadConfig(changed));
+                        Config config = new ConfigBuilder().build();
```

Comment:
dont load a particular kubeconf. Have the kubernetes-client load and merge all the files that are configured.